### PR TITLE
Use `drop_operation_state` to avoid stack overflows

### LINKED ISSUE #665: Explicitly release operation states early to release tiles for continuations
If we end up doing 1. or 2. from https://github.com/pika-org/pika/issues/479 we need to make changes in DLA-Future. Currently we rely on the pika implementation to release operation states/tiles early. The P2300 reference implementation does not release operation states early. This issue is meant to track whatever change we end up needing to do. I will most likely first go with an explicit sender adaptor that allows releasing the operation state early in tactical places.

### LINKED ISSUE #1005: Fix and re-enable stdexec debug tests in CI after #1003
Failures reported:
https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4700071344751697/7514005670787789/-/jobs/5239237411
https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4700071344751697/7514005670787789/-/jobs/5239298648

### PR DESCRIPTION
**Pull request by aurianer**
Will need the pika 0.19.1 patch release (it will probably be released next week), in the meantime I'm temporarily using pika@main in CI
Attempt to fix #1005. Fixes #665.

### CODE REVIEW

**Comment by msimberg** [comment]
cscs-ci run

File: CMakeLists.txt [hunk 1]
```diff
@@ -171,7 +171,7 @@ else()
 endif()
 
 # ----- pika
-find_package(pika 0.18.0 REQUIRED)
+find_package(pika REQUIRED)
```
Comment (rasolca) [inline CMakeLists.txt]: Please add version constrain back.

File: spack/packages/dla-future/package.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("pika@0.15.1:", when="@0.1")
     depends_on("pika@0.16:", when="@0.2.0")
     depends_on("pika@0.17:", when="@0.2.1:")
-    depends_on("pika@0.18:", when="@master")
+    depends_on("pika@main", when="@master")
```
Comment (rasolca) [inline spack/packages/dla-future/package.py]: still depend on `@main`

**Review by rasolca — COMMENTED** [review]
This PR still depend on `pika@main`. I think it should be updated to `pika@0.19.1` before considering it ready.

**Review by msimberg — COMMENTED** [review]
Before approving I'd like to hear if you were able to reproduce the bug locally and if the previously failing tests have been run enough times in CI? It was after all the `band_to_tridiag` miniapp that was failing that revealed the stack overflows, and now it was the C API eigensolver tests...

File: include/dlaf/sender/transform.h [hunk 3]
```diff
@@ -61,7 +62,8 @@ template <TransformDispatchType Tag = TransformDispatchType::Plain, Backend B =
   using dlaf::common::internal::Unwrapping;
 
   if constexpr (B == Backend::MC) {
-    return then(std::move(transfer_sender), ConsumeRvalues{Unwrapping{std::forward<F>(f)}});
+    return then(std::move(transfer_sender), ConsumeRvalues{Unwrapping{std::forward<F>(f)}}) |
+           drop_operation_state();
   }
   else if constexpr (B == Backend::GPU) {
 #if defined(DLAF_WITH_GPU)
```
Comment (msimberg) [inline include/dlaf/sender/transform.h:69]: I'd also like to see `drop_operation_state` used in the GPU branch(es) below.

**Review by msimberg — CHANGES_REQUESTED** [review]
This also needs a bump to the spack commit. It's currently installing `pika@main`.

**Comment by msimberg** [comment]
cscs-ci run

**Pull request merged**